Competition2017/src/competition/subsystems/drive/DriveSubsystem.java

```diff
@@ -5,37 +5,160 @@
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 
+import com.ctre.CANTalon.FeedbackDevice;
+import com.ctre.CANTalon.TalonControlMode;
 import xbot.common.command.BaseSubsystem;
-import xbot.common.controls.actuators.XSpeedController;
+import xbot.common.controls.actuators.XCANTalon;
 import xbot.common.injection.wpi_factories.WPIFactory;
+import xbot.common.math.MathUtils;
+import xbot.common.properties.DoubleProperty;
 import xbot.common.properties.XPropertyManager;
 
 @Singleton
 public class DriveSubsystem extends BaseSubsystem {
     private static Logger log = Logger.getLogger(DriveSubsystem.class);
 
-    public final XSpeedController leftFrontDrive;
-    public final XSpeedController leftRearDrive;
-    public final XSpeedController rightFrontDrive;
-    public final XSpeedController rightRearDrive;
+    public final XCANTalon leftDrive;
+    public final XCANTalon leftDriveSlave;
+    public final XCANTalon rightDrive;
+    public final XCANTalon rightDriveSlave;
+    
+    private final DoubleProperty encoderCodesProperty;
+    private final DoubleProperty maxSpeedProperty;
+    private final DoubleProperty p;
+    private final DoubleProperty i;
+    private final DoubleProperty d;
+    private final DoubleProperty f;
 
     @Inject
     public DriveSubsystem(WPIFactory factory, XPropertyManager propManager) {
         log.info("Creating DriveSubsystem");
 
-        this.leftFrontDrive = factory.getSpeedController(0);
-        this.leftRearDrive = factory.getSpeedController(2);
+        // TODO: Update these defaults. The current values are blind guesses.
+        encoderCodesProperty = propManager.createPersistentProperty("Drive encoder codes per rev", 512);
+        maxSpeedProperty = propManager.createPersistentProperty("Max drive motor speed (rotations per second)", 5);
 
-        this.rightFrontDrive = factory.getSpeedController(1);
-        this.rightFrontDrive.setInverted(true);
-        this.rightRearDrive = factory.getSpeedController(3);
-        this.rightRearDrive.setInverted(true);
+        p = propManager.createPersistentProperty("Drive P", 2);
```

Comment:
Since we'll likely be in a world where we have both position and velocity PID for the drive might be good to specify that here too with the name: `"Drive Vel P"` or something?